Sources/Classes/RSDeviceModeManager.m

```diff
@@ -37,14 +46,14 @@ - (void) startDeviceModeProcessor:(NSArray<RSServerDestination*>*) destinations
     [RSLogger logDebug:@"RSDeviceModeManager: DeviceModeProcessor: Initializing the Custom Factories"];
     [self initiateCustomFactories];
     [RSLogger logDebug:@"RSDeviceModeManager: DeviceModeProcessor: Replaying the message queue to the Factories"];
-    [self replayMessageQueue];
     self->areFactoriesInitialized = YES;
```

Comment:
We need to move this line to after the `else` branch because, we want the replay to happen first and then the `makeFactoryDump` to accept events

Reply:
I actually took this inspiration from the Android [PR](https://github.com/rudderlabs/rudder-sdk-android/blob/fix/events-not-removed-deba/core/src/main/java/com/rudderstack/android/sdk/core/RudderDeviceModeManager.java). 

Should we consider moving `self->areFactoriesInitialized = YES;` after [this](https://github.com/rudderlabs/rudder-sdk-ios/blob/510c8db7c360a428b8be8567020eb81663badc96/Sources/Classes/RSDeviceModeManager.m#L56) line and also add in the [if](https://github.com/rudderlabs/rudder-sdk-ios/blob/510c8db7c360a428b8be8567020eb81663badc96/Sources/Classes/RSDeviceModeManager.m#L51) block? It'll help in achieving our goal to replay the event first and then `makeFactoryDump` will accept the events? WDYT?

Reply:
Update: I'll revert this change, as decided in another comment, refer [here](https://github.com/rudderlabs/rudder-sdk-ios/pull/340#discussion_r1227527620).